src/baldr/incident_singleton.h

```diff
@@ -203,43 +216,64 @@ struct incident_singleton_t {
+          break;
+        }
+
         // check all of the timestamps/tile_ids
         for (auto entry : *changelog) {
-          // check the timestamp part of the tile id to see if its newer
+          // first 25 bits are the tile id
+          valhalla::baldr::GraphId tile_id(((uint64_t(1) << 25) - 1) & entry);
+          seen.insert(tile_id);
           // spare last 39 bits are the timestamp, leaves us with something like 17k years
           uint64_t timestamp = (entry >> 25) & ((uint64_t(1) << 39) - 1);
+          // check the timestamp part of the tile id to see if its newer
           if (last_scan <= timestamp) {
-            // first 25 bits are the tile id
-            valhalla::baldr::GraphId tile_id(((uint64_t(1) << 46) - 1) & entry);
             // concoct a file name from the tile_id
             auto file_location = inc_log_path;
             file_location.replace_filename(
                 valhalla::baldr::GraphTile::FileSuffix(tile_id, ".pbf", true));
             // update the tile
-            update_count += update_tile(state, tile_id, file_location.string());
+            update_count += update_tile(state, tile_id, read_tile(file_location.string()));
           }
         }
       } // we are in directory scan mode
       else if (!inc_dir.string().empty()) {
         // check all of the files
         for (filesystem::recursive_directory_iterator i(inc_dir), end; i != end; ++i) {
           try {
-            // if the tile was updated since the last time we scanned we load it
-            struct stat s;
+            // if this looks like a tile
             valhalla::baldr::GraphId tile_id;
             if (i->is_regular_file() &&
-                (tile_id = valhalla::baldr::GraphTile::GetTileId(i->path().string())).Is_Valid() &&
-                stat(i->path().c_str(), &s) == 0 && last_scan <= MTIME(s)) {
-              // update the tile
-              update_count += update_tile(state, tile_id, i->path().string());
+                (tile_id = valhalla::baldr::GraphTile::GetTileId(i->path().string())).Is_Valid()) {
+              // and if the tile was updated since the last time we scanned we load it
+              seen.insert(tile_id);
+              struct stat s;
+              if (stat(i->path().c_str(), &s) == 0 && last_scan <= MTIME(s)) {
+                // update the tile
+                update_count += update_tile(state, tile_id, read_tile(i->path().string()));
+              }
             }
-          } catch (...) { LOG_WARN("Incident watcher ignoring " + i->path().string()); }
+          } // happens when there is a file in the directory that doesnt have a tile-looking name
+          catch (...) {}
+        }
+      }
+
+      // for all the ones we didnt see, they have been removed from the filesystem or changelog
+      // no locking is needed because we don't realloc here we just null out some values
+      for (auto entry = state->cache.begin(); entry != state->cache.end(); ++entry) {
```

Comment:
here we go through and remove any tiles that went missing in the last round

Reply:
Which in the healthy case should not happen, since tiles that had incidents before but now have 0 incidents, still update the file on disk with an empty tile.